authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
See the note about an API break - I didn't want to break the public API, and this needed the same type as addAuthHeader which is the other implementation of adding authentication to a request, and addAuthHeader does use the first parameter.